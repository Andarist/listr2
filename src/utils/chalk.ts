import chalkDefault from 'chalk'

let chalk: chalkDefault.Chalk

/* istanbul ignore if */
if (process.env?.LISTR_DISABLE_COLOR !== '1' && chalkDefault.supportsColor) {
  chalk = new chalkDefault.Instance()
} else {
  chalk = new chalkDefault.Instance({ level: 0 })
}

/** A custom chalk instance to disable coloring compeletely. */
export default chalk
