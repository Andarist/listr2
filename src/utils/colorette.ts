import { createColors } from 'colorette'

export default createColors({ useColor: process.env?.LISTR_DISABLE_COLOR !== '1' })
