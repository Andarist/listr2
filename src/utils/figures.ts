import { isUnicodeSupported } from './is-unicode-supported'

const FIGURES_MAIN = {
  warning: '⚠',
  cross: '✖',
  arrowDown: '↓',
  tick: '✔',
  arrowRight: '→',
  pointer: '❯',
  checkboxOn: '☒',
  arrowLeft: '←',
  squareSmallFilled: '◼',
  pointerSmall: '›'
}

const FIGURES_FALLBACK = {
  ...FIGURES_MAIN,
  warning: '‼',
  cross: '×',
  tick: '√',
  pointer: '>',
  checkboxOn: '[×]',
  squareSmallFilled: '■'
}

export const figures = isUnicodeSupported() ? FIGURES_MAIN : FIGURES_FALLBACK
