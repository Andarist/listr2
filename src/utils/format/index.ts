export * from './colorette'
export * from './indent-string'
export * from './figures'
export * from './parse-time'
export * from './splat'
export * from './timestamp'
