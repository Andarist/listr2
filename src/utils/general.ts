import * as clone from 'clone'

/**
 * Deep clones a object in the most easiest manner.
 */
export function cloneObject<T extends Record<PropertyKey, any>> (obj: T): T {
  return clone(obj)
}
