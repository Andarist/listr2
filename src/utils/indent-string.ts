export function indentString (string: string, count: number): string {
  return string.replace(/^(?!\s*$)/gm, ' '.repeat(count))
}