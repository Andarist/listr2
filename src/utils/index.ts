export * from './environment'
export * from './format'
export * from './logger'
export * from './process-output'
export * from './ui'
export * from './assert'
export * from './clone'
export * from './concurrency'
export * from './delay'
export * from './uuid'
