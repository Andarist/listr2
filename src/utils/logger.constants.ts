/** Default loglevels for the logger */
export enum LogLevels {
  silent = 'silent',
  fail = 'fail',
  skip = 'skip',
  success = 'success',
  data = 'data',
  start = 'start',
  title = 'title',
  retry = 'retry',
  rollback = 'rollback'
}
