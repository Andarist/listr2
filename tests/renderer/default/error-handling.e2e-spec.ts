/* eslint-disable @typescript-eslint/no-empty-function */
import type { MockProcessOutput } from '@tests/utils'
import { expectProcessOutputToMatchSnapshot, mockProcessOutput, unmockProcessOutput } from '@tests/utils'

import { Listr } from '@root'

describe('default renderer: error handling', () => {
  const output: MockProcessOutput = {} as MockProcessOutput

  process.stdout.isTTY = true

  beforeEach(async () => {
    mockProcessOutput(output)
  })

  afterEach(async () => {
    unmockProcessOutput(output)
    jest.clearAllMocks()
  })

  // vxSr6uKuGeO4iX7r2j2Uo9SYlXRhz6nf
  it('should throw out an error', async () => {
    let err: Error

    try {
      await new Listr(
        [
          {
            title: 'This task will fail.',
            task: async (): Promise<void> => {
              throw new Error('This task failed after 2 seconds.')
            }
          },
          {
            title: 'This task will never execute.',
            task: (_, task): void => {
              task.title = 'I will change my title if this executes.'
            }
          }
        ],
        {
          concurrent: false,
          exitOnError: true,
          rendererOptions: { lazy: true }
        }
      ).run()
    } catch (e: any) {
      err = e
    }

    expect(err).toBeTruthy()
    expectProcessOutputToMatchSnapshot(output, 'vxSr6uKuGeO4iX7r2j2Uo9SYlXRhz6nf')
  })

  // OgQG0pbo8qgn5ZD8TLdlzM3HnnnYoozl
  it('should not throw out an error when exitonerror is disabled', async () => {
    let err: Error

    try {
      await new Listr(
        [
          {
            title: 'This task will fail.',
            task: async (): Promise<void> => {
              throw new Error('This task failed after 2 seconds.')
            }
          },
          {
            title: 'This task will execute.',
            task: (_, task): void => {
              task.title = 'I will change my title if this executes.'
            }
          }
        ],
        {
          concurrent: false,
          exitOnError: false,
          rendererOptions: { lazy: true }
        }
      ).run()
    } catch (e: any) {
      err = e
    }

    expect(err).toBeFalsy()
    expectProcessOutputToMatchSnapshot(output, 'OgQG0pbo8qgn5ZD8TLdlzM3HnnnYoozl')
  })

  // HUHzWgxW8O1FGgEisITyBb2eDKzmyhUO
  it('should change exit on error per subtask', async () => {
    let err: Error

    try {
      await new Listr(
        [
          {
            title: 'This task will execute and not quit on errors.',
            task: (_, task): Listr =>
              task.newListr(
                [
                  {
                    title: 'This is a subtask.',
                    task: async (): Promise<void> => {
                      throw new Error('I have failed [0]')
                    }
                  },
                  {
                    title: 'This is an another subtask.',
                    task: async (): Promise<void> => {
                      throw new Error('I have failed [1]')
                    }
                  },
                  {
                    title: 'This is yet an another subtask.',
                    task: async (_, task): Promise<void> => {
                      task.title = 'I have succeeded.'
                    }
                  }
                ],
                { exitOnError: false }
              )
          },
          {
            title: 'This task will execute.',
            task: (): void => {
              throw new Error('I will exit on error since I am a direct child of parent task.')
            }
          }
        ],
        {
          concurrent: false,
          exitOnError: true,
          rendererOptions: { lazy: true }
        }
      ).run()
    } catch (e: any) {
      err = e
    }

    expect(err).toBeTruthy()
    expectProcessOutputToMatchSnapshot(output, 'HUHzWgxW8O1FGgEisITyBb2eDKzmyhUO')
  })

  // H2KTg7q5F1kWMtrPFdOERVSZc3UT2IsM
  it('should contain errors from failed tasks', async () => {
    let err: Error
    const task: Listr<any, any, any> = new Listr(
      [
        {
          title: 'This task will execute and not quit on errors.',
          task: (_, task): Listr =>
            task.newListr(
              [
                {
                  title: 'This is a subtask.',
                  task: async (): Promise<void> => {
                    throw new Error('I have failed [0]')
                  }
                },
                {
                  title: 'This is an another subtask.',
                  task: async (): Promise<void> => {
                    throw new Error('I have failed [1]')
                  }
                },
                {
                  title: 'This is yet an another subtask.',
                  task: async (_, task): Promise<void> => {
                    task.title = 'I have succeeded.'
                  }
                }
              ],
              { exitOnError: false }
            )
        },
        {
          title: 'This task will execute.',
          task: (): void => {
            throw new Error('I will exit on error since I am a direct child of parent task.')
          }
        }
      ],
      {
        concurrent: false,
        exitOnError: true,
        rendererOptions: { lazy: true }
      }
    )

    try {
      await task.run()
    } catch (e: any) {
      err = e
    }

    expect(err).toBeTruthy()
    expect(task.err).toMatchSnapshot('H2KTg7q5F1kWMtrPFdOERVSZc3UT2IsM')
  })

  // 03IyrStkPGQBIcbYM0HQXoYQxDEVZu8H
  it('should throw out an error in the data field with collapse errors false', async () => {
    let err: Error

    try {
      await new Listr(
        [
          {
            title: 'This task will fail.',
            task: async (): Promise<void> => {
              throw new Error('This task failed after 2 seconds.')
            }
          },
          {
            title: 'This task will never execute.',
            task: (_, task): void => {
              task.title = 'I will change my title if this executes.'
            }
          }
        ],
        {
          concurrent: false,
          exitOnError: true,
          rendererOptions: { lazy: true, collapseErrors: false }
        }
      ).run()
    } catch (e: any) {
      err = e
    }

    expect(err).toBeTruthy()
    expectProcessOutputToMatchSnapshot(output, '03IyrStkPGQBIcbYM0HQXoYQxDEVZu8H')
  })

  // lnjpjmnHOxRSKy9J6YCMtqSAsVkHC3mH
  it('should show the default task title when failed with showErrorMessage of', async () => {
    let err: Error

    try {
      await new Listr(
        [
          {
            title: 'This task will fail.',
            task: async (): Promise<void> => {
              throw new Error('This task failed after 2 seconds.')
            }
          },
          {
            title: 'This task will never execute.',
            task: (_, task): void => {
              task.title = 'I will change my title if this executes.'
            }
          }
        ],
        {
          concurrent: false,
          exitOnError: true,
          rendererOptions: { lazy: true, showErrorMessage: false }
        }
      ).run()
    } catch (e: any) {
      err = e
    }

    expect(err).toBeTruthy()
    expectProcessOutputToMatchSnapshot(output, 'lnjpjmnHOxRSKy9J6YCMtqSAsVkHC3mH')
  })

  it.each([ true, false ])('should disable exitOnError from task level while: %s', async (exitOnError) => {
    let err: Error

    try {
      await new Listr(
        [
          {
            title: 'This task will fail.',
            task: async (): Promise<void> => {
              throw new Error('This task failed after 2 seconds.')
            },
            exitOnError
          },
          {
            title: 'This task will maybe execute.',
            task: (_, task): void => {
              task.title = 'I will change my title if this executes.'
            }
          }
        ],
        {
          concurrent: false,
          exitOnError: true,
          rendererOptions: { lazy: true }
        }
      ).run()
    } catch (e: any) {
      err = e
    }

    if (exitOnError) {
      expect(err).toBeTruthy()
    } else {
      expect(err).toBeFalsy()
    }

    expectProcessOutputToMatchSnapshot(output, 'Y9ADBbD3GX6P6HIKGTdeHybvNK4OPw37')
  })
})
