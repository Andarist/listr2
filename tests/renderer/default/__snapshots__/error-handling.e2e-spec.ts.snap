// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default renderer: error handling should change exit on error per subtask: HUHzWgxW8O1FGgEisITyBb2eDKzmyhUO-exit 1`] = `[]`;

exports[`default renderer: error handling should change exit on error per subtask: HUHzWgxW8O1FGgEisITyBb2eDKzmyhUO-stderr 1`] = `[]`;

exports[`default renderer: error handling should change exit on error per subtask: HUHzWgxW8O1FGgEisITyBb2eDKzmyhUO-stdout 1`] = `
[
  "‚óº This task will execute and not quit on errors.
",
  "[2K[1A[2K[G‚óº This task will execute and not quit on errors.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚óº This is a subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚óº This is a subtask.
  ‚óº This is an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚óº This is a subtask.
  ‚óº This is an another subtask.
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚ùØ This is a subtask.
  ‚óº This is an another subtask.
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ This is a subtask.
  ‚óº This is an another subtask.
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚óº This is an another subtask.
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚ùØ This is an another subtask.
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ This is an another subtask.
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚óº This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚ùØ This is yet an another subtask.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚ùØ I have succeeded.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ùØ This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚úî I have succeeded.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ö† This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚úî I have succeeded.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ö† This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚úî I have succeeded.
‚ùØ This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ö† This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚úî I have succeeded.
‚úñ This task will execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G‚ö† This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚úî I have succeeded.
‚úñ I will exit on error since I am a direct child of parent task.
",
  "[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G",
  "‚ö† This task will execute and not quit on errors.
  ‚úñ I have failed [0]
  ‚úñ I have failed [1]
  ‚úî I have succeeded.
‚úñ I will exit on error since I am a direct child of parent task.
",
]
`;

exports[`default renderer: error handling should contain errors from failed tasks: H2KTg7q5F1kWMtrPFdOERVSZc3UT2IsM 1`] = `
[
  [ListrError: I have failed [0]],
  [ListrError: I have failed [1]],
  [ListrError: I will exit on error since I am a direct child of parent task.],
]
`;

exports[`default renderer: error handling should disable exitOnError from task level while: false: Y9ADBbD3GX6P6HIKGTdeHybvNK4OPw37-exit 1`] = `[]`;

exports[`default renderer: error handling should disable exitOnError from task level while: false: Y9ADBbD3GX6P6HIKGTdeHybvNK4OPw37-stderr 1`] = `[]`;

exports[`default renderer: error handling should disable exitOnError from task level while: false: Y9ADBbD3GX6P6HIKGTdeHybvNK4OPw37-stdout 1`] = `
[
  "‚óº This task will fail.
",
  "[2K[1A[2K[G‚óº This task will fail.
‚óº This task will maybe execute.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will fail.
‚óº This task will maybe execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task will fail.
‚óº This task will maybe execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task failed after 2 seconds.
‚óº This task will maybe execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task failed after 2 seconds.
‚óº I will change my title if this executes.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task failed after 2 seconds.
‚úî I will change my title if this executes.
",
  "[2K[1A[2K[1A[2K[G",
  "‚úñ This task failed after 2 seconds.
‚úî I will change my title if this executes.
",
]
`;

exports[`default renderer: error handling should disable exitOnError from task level while: true: Y9ADBbD3GX6P6HIKGTdeHybvNK4OPw37-exit 1`] = `[]`;

exports[`default renderer: error handling should disable exitOnError from task level while: true: Y9ADBbD3GX6P6HIKGTdeHybvNK4OPw37-stderr 1`] = `[]`;

exports[`default renderer: error handling should disable exitOnError from task level while: true: Y9ADBbD3GX6P6HIKGTdeHybvNK4OPw37-stdout 1`] = `
[
  "‚óº This task will fail.
",
  "[2K[1A[2K[G‚óº This task will fail.
‚óº This task will maybe execute.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will fail.
‚óº This task will maybe execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task will fail.
‚óº This task will maybe execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task failed after 2 seconds.
‚óº This task will maybe execute.
",
  "[2K[1A[2K[1A[2K[G",
  "‚úñ This task failed after 2 seconds.
‚óº This task will maybe execute.
",
]
`;

exports[`default renderer: error handling should not throw out an error when exitonerror is disabled: OgQG0pbo8qgn5ZD8TLdlzM3HnnnYoozl-exit 1`] = `[]`;

exports[`default renderer: error handling should not throw out an error when exitonerror is disabled: OgQG0pbo8qgn5ZD8TLdlzM3HnnnYoozl-stderr 1`] = `[]`;

exports[`default renderer: error handling should not throw out an error when exitonerror is disabled: OgQG0pbo8qgn5ZD8TLdlzM3HnnnYoozl-stdout 1`] = `
[
  "‚óº This task will fail.
",
  "[2K[1A[2K[G‚óº This task will fail.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will fail.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task will fail.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task failed after 2 seconds.
‚óº This task will execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task failed after 2 seconds.
‚ùØ This task will execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task failed after 2 seconds.
‚ùØ I will change my title if this executes.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task failed after 2 seconds.
‚úî I will change my title if this executes.
",
  "[2K[1A[2K[1A[2K[G",
  "‚úñ This task failed after 2 seconds.
‚úî I will change my title if this executes.
",
]
`;

exports[`default renderer: error handling should show the default task title when failed with showErrorMessage of: lnjpjmnHOxRSKy9J6YCMtqSAsVkHC3mH-exit 1`] = `[]`;

exports[`default renderer: error handling should show the default task title when failed with showErrorMessage of: lnjpjmnHOxRSKy9J6YCMtqSAsVkHC3mH-stderr 1`] = `[]`;

exports[`default renderer: error handling should show the default task title when failed with showErrorMessage of: lnjpjmnHOxRSKy9J6YCMtqSAsVkHC3mH-stdout 1`] = `
[
  "‚óº This task will fail.
",
  "[2K[1A[2K[G‚óº This task will fail.
‚óº This task will never execute.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will fail.
‚óº This task will never execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task will fail.
‚óº This task will never execute.
",
  "[2K[1A[2K[1A[2K[G",
  "‚úñ This task will fail.
‚óº This task will never execute.
",
]
`;

exports[`default renderer: error handling should throw out an error in the data field with collapse errors false: 03IyrStkPGQBIcbYM0HQXoYQxDEVZu8H-exit 1`] = `[]`;

exports[`default renderer: error handling should throw out an error in the data field with collapse errors false: 03IyrStkPGQBIcbYM0HQXoYQxDEVZu8H-stderr 1`] = `[]`;

exports[`default renderer: error handling should throw out an error in the data field with collapse errors false: 03IyrStkPGQBIcbYM0HQXoYQxDEVZu8H-stdout 1`] = `
[
  "‚óº This task will fail.
",
  "[2K[1A[2K[G‚óº This task will fail.
‚óº This task will never execute.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will fail.
‚óº This task will never execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task will fail.
‚óº This task will never execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task will fail.
  ‚Ä∫ This task failed after 2 seconds.
‚óº This task will never execute.
",
  "[2K[1A[2K[1A[2K[1A[2K[G",
  "‚úñ This task will fail.
  ‚Ä∫ This task failed after 2 seconds.
‚óº This task will never execute.
",
]
`;

exports[`default renderer: error handling should throw out an error: vxSr6uKuGeO4iX7r2j2Uo9SYlXRhz6nf-exit 1`] = `[]`;

exports[`default renderer: error handling should throw out an error: vxSr6uKuGeO4iX7r2j2Uo9SYlXRhz6nf-stderr 1`] = `[]`;

exports[`default renderer: error handling should throw out an error: vxSr6uKuGeO4iX7r2j2Uo9SYlXRhz6nf-stdout 1`] = `
[
  "‚óº This task will fail.
",
  "[2K[1A[2K[G‚óº This task will fail.
‚óº This task will never execute.
",
  "[2K[1A[2K[1A[2K[G‚ùØ This task will fail.
‚óº This task will never execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task will fail.
‚óº This task will never execute.
",
  "[2K[1A[2K[1A[2K[G‚úñ This task failed after 2 seconds.
‚óº This task will never execute.
",
  "[2K[1A[2K[1A[2K[G",
  "‚úñ This task failed after 2 seconds.
‚óº This task will never execute.
",
]
`;
